pkg/collector/corechecks/servicediscovery/module/ignore_proc.go

```diff
@@ -0,0 +1,51 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+//go:build linux
+
+package module
+
+import (
+	"github.com/shirou/gopsutil/v3/process"
+)
+
+// ignoreServices is a list of service names that should not be reported as a service.
+var ignoreServices = map[string]struct{}{
+	"datadog-agent": {},
+}
```

Comment:
why only datadog agent?

Reply:
This is meantime only service defined in score of the ticket to be bypassed.

Reply:
it is misleading to have methods of the same struct defined on multiple files
let's fold it back to the original file

Reply:
The PR is pretty much irrelevant
Customers with the datadog agent don't set DD_SERVICE for us, and we don't do that by default

So I'm concerned the current implementation does not provide any actual value and impact

Reply:
Do you want to delete this PR and close Jira ticket?

Reply:
no, I want to improve the ticket description, and make the implementation better for the general case

Reply:
I've updated Jira ticket and title of this PR. Now this PR allows to add excluded service name to system-probe config. The config has one default service "datadog-agent"